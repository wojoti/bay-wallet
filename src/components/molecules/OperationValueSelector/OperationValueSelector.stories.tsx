import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {OperationValueSelector} from 'components/molecules';

export default {
  title: 'Molecules/OperationValueSelector',
  component: OperationValueSelector,
  argTypes: {
    onValuePress: {action: 'onValuePress'},
  },
} as ComponentMeta<typeof OperationValueSelector>;
const Template: ComponentStory<typeof OperationValueSelector> = args => (
  <OperationValueSelector {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
  value: '50',
  description: 'Operation Amount',
  selectedButton: 0,
};
