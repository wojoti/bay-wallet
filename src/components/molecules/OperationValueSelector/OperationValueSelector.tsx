import {Button, Container, Header, Label} from 'components/atoms';
import {PropsWithChildren} from 'react';
import style from './OperationValueSelector.style';

export type OperationValueSelectorProps = PropsWithChildren<{
  value: string;
  onValuePress: (value: string, id: number) => void;
  description: string;
  selectedButton: number;
  testId?: string;
}>;

const OperationValueSelector = ({
  value,
  onValuePress,
  selectedButton,
  description,
  testId,
}: OperationValueSelectorProps) => {
  const onValuePress50 = () => {
    onValuePress('50', 0);
  };

  const onValuePress100 = () => {
    onValuePress('100', 1);
  };

  const onValuePress200 = () => {
    onValuePress('200', 2);
  };

  const onValuePress400 = () => {
    onValuePress('400', 3);
  };

  const onValuePress700 = () => {
    onValuePress('700', 4);
  };
  const onValuePress1000 = () => {
    onValuePress('1.000', 5);
  };

  return (
    <Container customStyle={style.wrapper} testId={testId} align="center">
      <Container align="center">
        <Container
          customStyle={style.top}
          align="center"
          justify="space-between">
          <Header testId="operationvalueselector-header" size={30}>
            {value}
          </Header>
          <Label testId="operationvalueselector-label" color="grey" size={15}>
            {description}
          </Label>
        </Container>

        <Container
          customStyle={style.topBtnRow}
          direction="row"
          justify="space-between">
          <Button
            onPress={onValuePress50}
            customStyle={
              selectedButton === 0 ? style.pressedButton : style.button
            }
            customFontStyle={
              selectedButton === 0 ? style.pressedButtonFont : style.buttonFont
            }
            testId={'operationvalueselector-button-50'}>
            50
          </Button>

          <Button
            onPress={onValuePress100}
            customStyle={
              selectedButton === 1 ? style.pressedButton : style.button
            }
            customFontStyle={
              selectedButton === 1 ? style.pressedButtonFont : style.buttonFont
            }
            testId={'operationvalueselector-button-100'}>
            100
          </Button>

          <Button
            onPress={onValuePress200}
            customStyle={
              selectedButton === 2 ? style.pressedButton : style.button
            }
            customFontStyle={
              selectedButton === 2 ? style.pressedButtonFont : style.buttonFont
            }
            testId={'operationvalueselector-button-200'}>
            200
          </Button>
        </Container>
        <Container direction="row" justify="space-between">
          <Button
            onPress={onValuePress400}
            customStyle={
              selectedButton === 3 ? style.pressedButton : style.button
            }
            customFontStyle={
              selectedButton === 3 ? style.pressedButtonFont : style.buttonFont
            }
            testId={'operationvalueselector-button-400'}>
            400
          </Button>

          <Button
            onPress={onValuePress700}
            customStyle={
              selectedButton === 4 ? style.pressedButton : style.button
            }
            customFontStyle={
              selectedButton === 4 ? style.pressedButtonFont : style.buttonFont
            }
            testId={'operationvalueselector-button-700'}>
            700
          </Button>

          <Button
            onPress={onValuePress1000}
            customStyle={
              selectedButton === 5 ? style.pressedButton : style.button
            }
            customFontStyle={
              selectedButton === 5 ? style.pressedButtonFont : style.buttonFont
            }
            testId={'operationvalueselector-button-1000'}>
            1.000
          </Button>
        </Container>
      </Container>
    </Container>
  );
};

export default OperationValueSelector;
