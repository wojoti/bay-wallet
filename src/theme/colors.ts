export const lightPrimary = '#4E439E';
export const primary = '#363062';
export const secondary = 'white';

export const darkGrey = '#595959';
export const grey = '#6B6B6B';
export const vlightGrey = '#D1D1D1';
export const borderGrey = '#E2E2E2';

export const iconGrey = '#BDBDBD';

export const searchGrey = '#F5F5F5';

export const separator = '#F6F6F6';

export const green = '#61C277';
export const lime = '#53E520';
export const red = '#E52020';

export const successGreen = '#3EB290';

export const pink = '#D8B9C3';
export const darkPurple = '#26204E';
export const balanceWhite = '#F8F8FF';

export const marketDarkGrey = '#3D454B';

export const cardYellow = '#FFEF60';

export const buttonGrey = '#EFEFEF';
