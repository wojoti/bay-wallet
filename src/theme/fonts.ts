export const font = 'Poppins-Light';
export const boldFont = 'Poppins-Bold';
