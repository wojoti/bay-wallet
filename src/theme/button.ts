import {StyleSheet} from 'react-native';
import {primary, secondary} from './colors';
const buttonStyle = StyleSheet.create({
  primary: {
    borderWidth: 0,
    backgroundColor: primary,
  },
  secondary: {
    borderColor: primary,
    backgroundColor: secondary,
  },
});

export const buttonTextStyle = StyleSheet.create({
  primary: {
    color: secondary,
  },
  secondary: {
    color: primary,
  },
});

export default buttonStyle;
