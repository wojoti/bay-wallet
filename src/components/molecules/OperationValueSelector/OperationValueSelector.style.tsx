import {StyleSheet} from 'react-native';
import {buttonGrey, grey, primary, secondary} from 'theme/colors';

const style = StyleSheet.create({
  wrapper: {
    backgroundColor: secondary,
  },
  top: {
    height: 70,
    marginBottom: 40,
  },
  button: {
    borderWidth: 0,
    width: 95,
    height: 45,
    backgroundColor: buttonGrey,
  },
  buttonFont: {
    color: grey,
    fontSize: 19,
  },

  pressedButton: {
    borderWidth: 0,
    width: 95,
    height: 45,
    backgroundColor: primary,
  },
  pressedButtonFont: {
    color: secondary,
    fontSize: 19,
  },

  topBtnRow: {
    marginBottom: 20,
  },
});
export default style;
