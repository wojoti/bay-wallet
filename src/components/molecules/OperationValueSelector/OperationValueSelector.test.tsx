import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import {buttonGrey, primary} from 'theme/colors';
import OperationValueSelector, {
  OperationValueSelectorProps,
} from './OperationValueSelector';

let value = '$ 50';
let id = 0;
const mockOnValuePress = jest.fn((fnValue, fnId) => {
  value = '$ ' + fnValue;
  id = fnId;
});
const props: OperationValueSelectorProps = {
  testId: 'test-operationvalueselector-id',
  value: value,
  onValuePress: mockOnValuePress,
  description: 'Operation Amount',
  selectedButton: id,
};

const additionalProps: OperationValueSelectorProps = {
  ...props,
  selectedButton: 3,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(
    <OperationValueSelector {...props} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render OperationValueSelector', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
});

test('should render OperationValueSelector - header', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const headerElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-header',
  );
  expect(headerElement).toBeOnTheScreen();
  expect(headerElement).toHaveTextContent('$ 50');
});

test('should render OperationValueSelector - label', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const labelElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-label',
  );
  expect(labelElement).toBeOnTheScreen();
  expect(labelElement).toHaveTextContent('Operation Amount');
});

test('should render OperationValueSelector - button - 50', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-50',
  );
  expect(buttonElement).toBeOnTheScreen();
  const textElement = within(buttonElement).getByTestId('button-text');
  expect(textElement).toHaveTextContent('50');
});

test('should render OperationValueSelector - button - 100', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-100',
  );
  expect(buttonElement).toBeOnTheScreen();
  const textElement = within(buttonElement).getByTestId('button-text');
  expect(textElement).toHaveTextContent('100');
});

test('should render OperationValueSelector - button - 200', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-200',
  );
  expect(buttonElement).toBeOnTheScreen();
  const textElement = within(buttonElement).getByTestId('button-text');
  expect(textElement).toHaveTextContent('200');
});
test('should render OperationValueSelector - button - 400', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-400',
  );
  expect(buttonElement).toBeOnTheScreen();
  const textElement = within(buttonElement).getByTestId('button-text');
  expect(textElement).toHaveTextContent('400');
});
test('should render OperationValueSelector - button - 700', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-700',
  );
  expect(buttonElement).toBeOnTheScreen();
  const textElement = within(buttonElement).getByTestId('button-text');
  expect(textElement).toHaveTextContent('700');
});
test('should render OperationValueSelector - button - 1000', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-1000',
  );
  expect(buttonElement).toBeOnTheScreen();
  const textElement = within(buttonElement).getByTestId('button-text');
  expect(textElement).toHaveTextContent('1.000');
});

test('should render OperationValueSelector - button - 50 - handle onPress', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-50',
  );
  expect(buttonElement).toBeOnTheScreen();
  const textElement = within(buttonElement).getByTestId('button-text');
  expect(textElement).toHaveTextContent('50');
  fireEvent.press(buttonElement);
  expect(mockOnValuePress).toHaveBeenLastCalledWith('50', 0);

  expect(buttonElement).toHaveStyle({backgroundColor: primary});

  const headerElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-header',
  );
  expect(headerElement).toBeOnTheScreen();
  expect(headerElement).toHaveTextContent('$ 50');
});

test('should render OperationValueSelector - button - 100 - handle onPress', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-100',
  );
  expect(buttonElement).toBeOnTheScreen();
  const textElement = within(buttonElement).getByTestId('button-text');
  expect(textElement).toHaveTextContent('100');

  fireEvent.press(buttonElement);
  expect(mockOnValuePress).toHaveBeenLastCalledWith('100', 1);
});

test('should render OperationValueSelector - button - 200 - handle onPress', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-200',
  );
  expect(buttonElement).toBeOnTheScreen();
  const textElement = within(buttonElement).getByTestId('button-text');
  expect(textElement).toHaveTextContent('200');

  fireEvent.press(buttonElement);
  expect(mockOnValuePress).toHaveBeenLastCalledWith('200', 2);
});
test('should render OperationValueSelector - button - 400 - handle onPress', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-400',
  );
  expect(buttonElement).toBeOnTheScreen();
  const textElement = within(buttonElement).getByTestId('button-text');
  expect(textElement).toHaveTextContent('400');

  fireEvent.press(buttonElement);
  expect(mockOnValuePress).toHaveBeenLastCalledWith('400', 3);
});
test('should render OperationValueSelector - button - 700 - handle onPress', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-700',
  );
  expect(buttonElement).toBeOnTheScreen();
  const textElement = within(buttonElement).getByTestId('button-text');
  expect(textElement).toHaveTextContent('700');

  fireEvent.press(buttonElement);
  expect(mockOnValuePress).toHaveBeenLastCalledWith('700', 4);
});
test('should render OperationValueSelector - button - 1000 - handle onPress', () => {
  render(<OperationValueSelector {...props} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  const buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-1000',
  );
  expect(buttonElement).toBeOnTheScreen();
  const textElement = within(buttonElement).getByTestId('button-text');
  expect(textElement).toHaveTextContent('1.000');

  fireEvent.press(buttonElement);
  expect(mockOnValuePress).toHaveBeenLastCalledWith('1.000', 5);
});

test('should render OperationValueSelector - werify selected button styling', () => {
  render(<OperationValueSelector {...additionalProps} />);
  const OperationValueSelectorElement = screen.getByTestId(
    'test-operationvalueselector-id',
  );
  expect(OperationValueSelectorElement).toBeOnTheScreen();
  let buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-50',
  );
  expect(buttonElement).toBeOnTheScreen();
  expect(buttonElement).toHaveStyle({backgroundColor: buttonGrey});

  buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-100',
  );
  expect(buttonElement).toBeOnTheScreen();
  expect(buttonElement).toHaveStyle({backgroundColor: buttonGrey});

  buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-200',
  );
  expect(buttonElement).toBeOnTheScreen();
  expect(buttonElement).toHaveStyle({backgroundColor: buttonGrey});

  buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-400',
  );
  expect(buttonElement).toBeOnTheScreen();
  expect(buttonElement).toHaveStyle({backgroundColor: primary});

  buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-700',
  );
  expect(buttonElement).toBeOnTheScreen();
  expect(buttonElement).toHaveStyle({backgroundColor: buttonGrey});

  buttonElement = within(OperationValueSelectorElement).getByTestId(
    'operationvalueselector-button-1000',
  );
  expect(buttonElement).toBeOnTheScreen();
  expect(buttonElement).toHaveStyle({backgroundColor: buttonGrey});
});
