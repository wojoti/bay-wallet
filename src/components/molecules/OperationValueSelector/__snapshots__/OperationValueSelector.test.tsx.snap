// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      {
        "backgroundColor": "white",
      },
    ]
  }
  testID="test-operationvalueselector-id"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "width": "85%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "space-between",
            "width": "100%",
          },
          {
            "height": 70,
            "marginBottom": 40,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontFamily": "Poppins-Bold",
            "fontSize": 30,
            "letterSpacing": undefined,
            "textAlign": "left",
          }
        }
        testID="operationvalueselector-header"
      >
        $ 50
      </Text>
      <Text
        style={
          {
            "color": "grey",
            "fontFamily": "Poppins-Light",
            "fontSize": 15,
            "fontWeight": "400",
            "textAlign": "left",
          }
        }
        testID="operationvalueselector-label"
      >
        Operation Amount
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": undefined,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
          {
            "marginBottom": 20,
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 65,
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "backgroundColor": "#363062",
              "borderWidth": 0,
              "height": 45,
              "width": 95,
            },
          ]
        }
        testID="operationvalueselector-button-50"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Bold",
                "fontSize": 18,
                "textAlign": "center",
              },
              {
                "color": "white",
                "fontSize": 19,
              },
            ]
          }
          testID="button-text"
        >
          50
        </Text>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 65,
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "backgroundColor": "#EFEFEF",
              "borderWidth": 0,
              "height": 45,
              "width": 95,
            },
          ]
        }
        testID="operationvalueselector-button-100"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Bold",
                "fontSize": 18,
                "textAlign": "center",
              },
              {
                "color": "#6B6B6B",
                "fontSize": 19,
              },
            ]
          }
          testID="button-text"
        >
          100
        </Text>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 65,
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "backgroundColor": "#EFEFEF",
              "borderWidth": 0,
              "height": 45,
              "width": 95,
            },
          ]
        }
        testID="operationvalueselector-button-200"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Bold",
                "fontSize": 18,
                "textAlign": "center",
              },
              {
                "color": "#6B6B6B",
                "fontSize": 19,
              },
            ]
          }
          testID="button-text"
        >
          200
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": undefined,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 65,
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "backgroundColor": "#EFEFEF",
              "borderWidth": 0,
              "height": 45,
              "width": 95,
            },
          ]
        }
        testID="operationvalueselector-button-400"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Bold",
                "fontSize": 18,
                "textAlign": "center",
              },
              {
                "color": "#6B6B6B",
                "fontSize": 19,
              },
            ]
          }
          testID="button-text"
        >
          400
        </Text>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 65,
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "backgroundColor": "#EFEFEF",
              "borderWidth": 0,
              "height": 45,
              "width": 95,
            },
          ]
        }
        testID="operationvalueselector-button-700"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Bold",
                "fontSize": 18,
                "textAlign": "center",
              },
              {
                "color": "#6B6B6B",
                "fontSize": 19,
              },
            ]
          }
          testID="button-text"
        >
          700
        </Text>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 65,
              "justifyContent": "center",
              "opacity": 1,
            },
            false,
            {
              "backgroundColor": "#EFEFEF",
              "borderWidth": 0,
              "height": 45,
              "width": 95,
            },
          ]
        }
        testID="operationvalueselector-button-1000"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Poppins-Bold",
                "fontSize": 18,
                "textAlign": "center",
              },
              {
                "color": "#6B6B6B",
                "fontSize": 19,
              },
            ]
          }
          testID="button-text"
        >
          1.000
        </Text>
      </View>
    </View>
  </View>
</View>
`;
